lesson_21/homework_21.py

```diff
@@ -0,0 +1,114 @@
+"""Homework 21."""
+
+import logging
+from datetime import datetime
+
+"""
+Моніторингова система клєнта надсилає сигнал,
+що вона працездатна кожні 30-31 сек
+- наприкладTimestamp 05:45:40, а в наступному повідомлені — Timestamp 05:45:09
+(тут різниця heartbeat в 31 секунду)
+
+Є декілька дублючих потоків, що шлють дані одночасно,
+тож ми можемо проаналізувати лише один потік - Key TSTFEED0300|7E3E|0400
+
+Засобами автоматизації проаналізуйте наданий нам лог: hblog.txt
+
+відберіть лише строки з вказаним ключем Key TSTFEED0300|7E3E|0400
+Створіть функцію, що поверне лог-файл, де буде аналіз правильності вимог:
+для кожного випадку де heartbeat більше 31 сек але
+менше 33 логувало WARNING в файл hb_test.log
+для кожного випадку де heartbeat більше рівно 33
+логувало ERROR в файл hb_test.log
+3.Зверніть увагу, що нам для аналізу помилок було
+б добре знати час, в який помилка відбулася.
+
+Обов’язково включіть результат роботи — файл hb_test.log в PR.
+"""
+
+
+class HBAnalyzer:
+    """Class to analyze heartbeat logs."""
+
+    def __init__(self, log_file, key, timestamp_prefix,
+                 warn_limit, err_limit, time_format='%H:%M:%S'):
+        """Initialize parameters."""
+        self.log_file = log_file
+        self.key = key
+        self.timestamp_prefix = timestamp_prefix
+        self.warn_limit = warn_limit
+        self.err_limit = err_limit
+        self.time_format = time_format
+        self.logger = self._configure_logger()
+
+    def _configure_logger(self):
+        """Configure logger."""
+        logger = logging.getLogger(__name__)
+        logger.setLevel(logging.INFO)
+        handler = logging.FileHandler('hb_test.log', mode='a')
```

Comment:
log file name better to define as init param too, maybe with default. No sense to use it as hardcode. Or as minimum place to constants